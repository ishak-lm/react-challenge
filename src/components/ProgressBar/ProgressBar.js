import React from 'react'
import PropTypes from 'prop-types'

import S from './ProgressBar.styles'

const ProgressBar = ({ width, color, right, left }) => (
  <S.ProgressBar width={width} color={color} right={right} left={left} />
)

ProgressBar.propTypes = {
  width: PropTypes.number,
  color: PropTypes.string,
  right: PropTypes.bool,
  left: PropTypes.bool,
}

ProgressBar.defaultProps = {
  width: 100,
  color: 'green',
  right: false,
  left: false,
}

export default ProgressBar
