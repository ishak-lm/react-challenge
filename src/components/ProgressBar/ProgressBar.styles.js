import styled from 'styled-components'

import { colors } from '../../constants'

const ProgressBar = styled.div`
  position: absolute;
  width: ${({ width }) => `${width}%`};
  height: 25px;
  background-color: ${({ color }) => `rgb(${colors.progressBar[color]})`};
  right: ${({ right }) => (right ? 0 : 'inherit')};
  left: ${({ left }) => (left ? 0 : 'inherit')};
  z-index: 0;
`

export default {
  ProgressBar,
}
