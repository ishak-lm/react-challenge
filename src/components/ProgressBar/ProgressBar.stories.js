import React from 'react'
import { withKnobs, boolean, select, number } from '@storybook/addon-knobs'

import ProgressBar from './ProgressBar'

export default {
  title: '01 - ProgressBar',
}

export const StoryProgressBar = () => (
  <ProgressBar
    width={number('Width', 12, { range: true, min: 0, max: 100 })}
    color={select('Color', ['green', 'red'])}
    right={boolean('Right', true)}
    left={boolean('Left', false)}
  />
)

StoryProgressBar.story = {
  decorators: [withKnobs],
  parameters: { options: { selectedPanel: 'storybook/knobs/panel' } },
}
