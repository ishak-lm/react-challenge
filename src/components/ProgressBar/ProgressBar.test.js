/* global describe it expect */
import React from 'react'
import { mount } from 'enzyme'

import ProgressBar from './ProgressBar'

describe('ProgressBar', () => {
  describe('Default Progressbar', () => {
    const wrapper = mount(<ProgressBar />)

    it('should have the correct default props', () => {
      expect(wrapper.props()).toMatchObject({
        width: 100,
        color: 'green',
        right: false,
        left: false,
      })
    })
  })

  describe('Progressbar with Props', () => {
    const wrapper = mount(<ProgressBar width={47} color="red" right />)

    expect(wrapper).toMatchSnapshot()

    it('should have the correct default props', () => {
      expect(wrapper.props()).toMatchObject({
        width: 47,
        color: 'red',
        right: true,
        left: false,
      })
    })
  })
})
