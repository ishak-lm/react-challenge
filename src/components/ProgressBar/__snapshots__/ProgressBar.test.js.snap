// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<ProgressBar
  color="red"
  left={false}
  right={true}
  width={47}
>
  <styled.div
    color="red"
    left={false}
    right={true}
    width={47}
  >
    <StyledComponent
      color="red"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "evQXqx",
            "rules": Array [
              "
  position: absolute;
  width: ",
              [Function],
              ";
  height: 25px;
  background-color: ",
              [Function],
              ";
  right: ",
              [Function],
              ";
  left: ",
              [Function],
              ";
  z-index: 0;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      left={false}
      right={true}
      width={47}
    >
      <div
        className="sc-bdVaJa evQXqx"
        color="red"
        width={47}
      />
    </StyledComponent>
  </styled.div>
</ProgressBar>
`;
